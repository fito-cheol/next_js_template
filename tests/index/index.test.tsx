import { render, screen } from '@testing-library/react';
import Home from '../../src/app/page';

describe('Home', () => {
  it('renders a heading', () => {
    const { container } = render(<Home />);

    const heading = screen.getByRole('heading', {
      name: /welcome to next\.js!/i,
    });
    // Next Template Home

    expect(heading).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
