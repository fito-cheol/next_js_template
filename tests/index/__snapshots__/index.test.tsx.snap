// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a heading 1`] = `
<div>
  <main
    class="main"
  >
    <div
      class="description"
    >
      <h1>
         welcome to next.js! 
      </h1>
      <a
        href="dashboard"
      >
         Dashboard 
      </a>
      <img
        alt="belica"
        data-nimg="1"
        decoding="async"
        height="180"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2FBelica.webp&w=384&q=75"
        srcset="/_next/image?url=%2Fimages%2FBelica.webp&w=256&q=75 1x, /_next/image?url=%2Fimages%2FBelica.webp&w=384&q=75 2x"
        style="color: transparent;"
        width="140"
      />
      <button
        class="button"
      >
        i like round
      </button>
    </div>
  </main>
</div>
`;
